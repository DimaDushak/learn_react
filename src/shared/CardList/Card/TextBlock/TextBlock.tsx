/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import styles from './textblock.css';
import UserBlock from '../../../UserBlock';

export default function TextBlock(): JSX.Element {
    return (
        <div className={styles.textBlock}>
            <div className={styles.authorInfoBlock}>
                <UserBlock
                    link="#user"
                    avatarSrc="https://cdn.dribbble.com/users/2202649/avatars/small/2044ffe8e8848aeb80e514229c21c515.png?1616573826"
                    userName="Константин Кодов"
                />
                <span className={styles.publishTime}>
                    <span className={styles.publish}>опубликовано</span>8 часов
                    назад
                </span>
            </div>
            <h2 className={styles.postHeading}>
                <a href="#post" className={styles.postLink}>
                    Следует отметить, что новая модель организационной
                    деятельности и деятельности и деятельности и деятельности
                </a>
            </h2>
        </div>
    );
}
