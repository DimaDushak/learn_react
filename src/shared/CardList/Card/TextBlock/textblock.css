.textBlock {
    display: flex;
    flex-direction: column;
    padding: 22px 20px 13px;
}

.authorInfoBlock {
    align-items: center;
    display: flex;
    margin-bottom: 7px;
}

.authorInfoBlock a {
    color: var(--orange);
}

.authorInfoBlock:hover a {
    text-decoration: underline;
}

.authorInfoBlock a span,
.authorInfoBlock a img {
    position: relative;
    z-index: 1;
}

.publishTime {
    color: var(--grey99);
    font-size: 0.5rem;
    line-height: 0.6rem;
    margin-left: 5px;
}

.publish {
    display: none;
}

.postHeading {
    font-size: 0.8rem;
    font-weight: 400;
    margin: 0;
}

.postLink {
    display: -webkit-box;
    overflow: hidden;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
}

.postLink:after {
    bottom: 0;
    content: '';
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
}

@media (min-width: 1024px) {
    .textBlock {
        flex-grow: 1;
        padding: 0 0 0 13px;
    }

    .authorInfoBlock {
        margin-bottom: 0;
    }

    .publishTime {
        font-size: 0.7rem;
        line-height: 0.8rem;
        margin-right: 7px;
        margin-left: 0;
        order: -1;
    }

    .publish {
        display: inline;
        margin-right: 4px;
    }

    .postHeading {
        font-size: 1rem;
        line-height: 1.2rem;
        margin-bottom: 10px;
        max-width: 540px;
        order: -1;
    }

    .postLink {
        -webkit-line-clamp: 1;
    }
}

@media (min-width: 1540px) {
    .postHeading {
        max-width: 672px;
    }
}
