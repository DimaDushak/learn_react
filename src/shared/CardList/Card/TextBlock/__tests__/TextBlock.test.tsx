import { render } from '@testing-library/react';
import React from 'react';
import TextBlock from '../TextBlock';

test('should render TextBlock (snapshot)', () => {
    const { container } = render(<TextBlock />);

    expect(container.firstChild).toMatchSnapshot();
});
