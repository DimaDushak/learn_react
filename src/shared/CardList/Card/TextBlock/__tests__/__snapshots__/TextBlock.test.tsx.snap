// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextBlock (snapshot) 1`] = `
<div
  class="textBlock"
>
  <div
    class="authorInfoBlock"
  >
    <a
      class="userBlock userBlockSizes"
      href="#user"
    >
      <img
        alt="#"
        class="authorAvatar"
        src="https://cdn.dribbble.com/users/2202649/avatars/small/2044ffe8e8848aeb80e514229c21c515.png?1616573826"
      />
      <span
        class=""
      >
        Константин Кодов
      </span>
    </a>
    <span
      class="publishTime"
    >
      <span
        class="publish"
      >
        опубликовано
      </span>
      8 часов назад
    </span>
  </div>
  <h2
    class="postHeading"
  >
    <a
      class="postLink"
      href="#post"
    >
      Следует отметить, что новая модель организационной деятельности и деятельности и деятельности и деятельности
    </a>
  </h2>
</div>
`;
